pkg/agent/multicast/mcast_controller.go

```diff
@@ -473,6 +479,113 @@ func (c *Controller) addOrUpdateGroupEvent(e *mcastGroupEvent) {
 	}
 }
 
+func (c *Controller) memberChanged(e interface{}) {
+	podEvent := e.(types.PodUpdate)
+	namespace, name := podEvent.PodNamespace, podEvent.PodName
+	containerID := podEvent.ContainerID
+
+	klog.V(2).InfoS("Calling memberChanged", "IsAdd", podEvent.IsAdd, "namespace", namespace,
+		"name", name)
+	if podEvent.IsAdd {
+		iface, ok := c.ifaceStore.GetContainerInterface(containerID)
+		if ok {
+			g := &mcastGroupEvent{
+				eType: groupJoin,
+				iface: iface,
+			}
+			c.queryGroupCh <- g
+		}
+	} else {
+		interfaceName := util.GenerateContainerInterfaceName(name, namespace, containerID)
+		ifaceConfig := &interfacestore.InterfaceConfig{
+			InterfaceName: interfaceName,
+		}
+		g := &mcastGroupEvent{
+			eType: groupLeave,
+			iface: ifaceConfig,
+		}
+		c.queryGroupCh <- g
+		c.removeLocalInterface(ifaceConfig)
+	}
+}
+
+func (c *Controller) syncQueryGroup(stopCh <-chan struct{}) {
+	if c.queryGroupId == 0 {
+		klog.V(2).InfoS("As c.queryGroupId is 0, query group will not be installed")
+		return
+	}
+	for {
+		klog.InfoS("Resync query group")
+		select {
+		case event := <-c.queryGroupCh:
```

Comment:
If the chan can only send event once, I thnik we should not share. For example, if a group "225.1.2.3" has a member join, it comes to queryGroup, we shall ignore the event. But syncGroup will not receive the event since the event is comsumed by queryGroup

Reply:
Sorry to misunderstand the comment, will share groupEventCh